compiler/src/dotty/tools/dotc/typer/Namer.scala

```diff
@@ -246,10 +247,13 @@ class Namer { typer: Typer =>
       case tree: TypeDef if tree.isClassDef =>
         val flags = checkFlags(tree.mods.flags)
         val name = checkNoConflict(tree.name, flags.is(Private), tree.span).asTypeName
+        val tastyVersion = (TastyFormat.MajorVersion.toLong << (28 * 2))
```

Comment:
possibly you could lift this implementation into a shared method (in TastyUnpickler object)?